lab/lab.go

```diff
@@ -208,24 +207,26 @@ func (l *lab) Restart(ctx context.Context) error {
 	return nil
 }
 
-func (l *lab) Close() error {
+func (l *lab) Close() error{
 	var wg sync.WaitGroup
-
-	closer := func(c io.Closer) {
-		if err := c.Close(); err != nil {
-			log.Warn().Msgf("error while closing lab: %s", err)
-		}
-		wg.Done()
-	}
-
-	if l.environment != nil {
-		wg.Add(1)
-		go closer(l.environment)
-	}
-
+	 for _, lab  := range l.frontends {
+	 	wg.Add(1)
+	 	go func (){
```

Comment:
I would suggest to pass the lab as a parameter to the goroutine function something like this:
```
go func (lab lab.Lab){
	// do things with lab
}(lab)
```

Reply:
What is the main reason of that ?

Reply:
discussed.